internal/discord/command-handlers.go

```diff
@@ -479,3 +479,96 @@ func (d *Discord) InteractionCreate(s *discordgo.Session, i *discordgo.Interacti
 		d.ShowAllWarnings(s, i)
 	}
 }
+
+// ExileCheckUserHelper checks whether the user meets the requirements to be exiled
+// returns nil if the user does not and returns the member on success
+func (d *Discord) ExileCheckUserHelper(state *InteractionState, userID string) (*discordgo.Member, error) {
+	// Check if user exists in guild
+	member, err := d.GetUserInGuild(state.interaction.GuildID, userID)
+	if err != nil {
+		tempstr := fmt.Sprintf("Could not find user <@%v> in guild", userID)
+		fmt.Printf("%v: %v\n", tempstr, err)
+		RespondToInteraction(state.session, state.interaction.Interaction, tempstr, &state.isFirst)
+		return nil, err
```

Comment:
let's check they already don't have the exile role

Reply:
Not checking for the roles in this block of code is intentional (related to this comment https://github.com/naurffxiv/moddingway/pull/12#discussion_r1632511083) it is done later down the line